src/graph/service/PermissionCheck.cpp

```diff
@@ -109,8 +109,19 @@ namespace graph {
     case Sentence::Kind::kUpdateEdge:
     case Sentence::Kind::kDeleteVertices:
     case Sentence::Kind::kDeleteTags:
-    case Sentence::Kind::kDeleteEdges:
+    case Sentence::Kind::kDeleteEdges: {
+      return PermissionManager::canWriteData(session, vctx);
+    }
     case Sentence::Kind::kAdminJob: {
+      auto *adminJobSentence = dynamic_cast<AdminJobSentence *>(sentence);
+      if (adminJobSentence == nullptr) {
+        // should not happend.
+        LOG(WARNING) << "sentence is not AdminJobSentence";
+        return Status::PermissionError("No permission to write space.");
+      }
+      if (adminJobSentence->needWriteSpace()) {
```

Comment:
what does  `needWriteSpace` mean?

Reply:
> what does `needWriteSpace` mean? Why only balance job need write space? others like compact will not write?

Yes, It's means space data distribuition level permission needed.( And it's the origin permission needed before refactor to jobmanager.

Reply:
you may need to update the comment before `PermissionCheck::permissionCheck` to explain this

Reply:
updated.